app/controllers/posts_controller.rb

```diff
@@ -241,6 +241,8 @@ def update
     topic = post.topic
     topic = Topic.with_deleted.find(post.topic_id) if guardian.is_staff?
 
+    opts[:increment_edits_count] = true
```

Comment:
This looks like a new option we added. I'm curious as to why we need it.

Reply:
Hey Alan, it was because of this. Do you suggest a better way to tackle this issue?
https://meta.discourse.org/t/edit-log-by-user-audition-project/194791/16

Reply:
The count revision needs to go through a central pipeline, either in a model callback or in this case the `PostRevisor`. Otherwise, the code may be calling `PostRevisor` from somewhere else and the count wouldn't be kept updated. 

I looked at the meta topic but it isn't clear to me what the topics controller was doing such that it was revising the post?

Reply:
Hey, it is because of this [condition](https://github.com/discourse/discourse/blob/master/app/controllers/topics_controller.rb#L377) 

```
if changes.length > 0
first_post = topic.ordered_posts.first
success = PostRevisor.new(first_post, topic).revise!(current_user, changes, validate_post: false)
```
In my case, the value of changes is `{:tags=>[]}`
I'm not sure if this is the intended behavior or if it is a bug. What do you think?

Reply:
I think this is a bug, if there are no changes in the number of tags, there shouldn't be an edit made.

Reply:
The value of this variable is around line 377
https://github.com/discourse/discourse/blob/main/app/controllers/topics_controller.rb#L377

This happens when you edit just the content of the initial post - even if there are no changes to the tags.
I'm debugging to see if I can fix this issue, I'll keep you posted but if you have any ideas let me know.